### PR TITLE
Fix LatexEmbeddedLexer

### PR DESCRIPTION
**Pull request by eric-wieser**
Closes gh-1516

### CODE REVIEW

File: pygments/formatters/latex.py [hunk 1]
```diff
@@ -446,30 +446,62 @@ def __init__(self, left, right, lang, **options):
...
-    def get_tokens_aux(self, index, text):
+    def _find_escape_tokens(self, text):
+        """ Find escape tokens within text, give token=None otherwise """
+        index = 0
         while text:
             a, sep1, text = text.partition(self.left)
             if a:
-                for i, t, v in self.lang.get_tokens_unprocessed(a):
-                    yield index + i, t, v
-                    index += len(a)
```
Comment (eric-wieser) [inline pygments/formatters/latex.py:472]: This line was a bug, it should have been outside the `for` loop

File: pygments/formatters/latex.py [hunk 2]
```diff
@@ -446,30 +446,62 @@ def __init__(self, left, right, lang, **options):
...
+
 
-    def get_tokens_aux(self, index, text):
+    def _find_escape_tokens(self, text):
+        """ Find escape tokens within text, give token=None otherwise """
+        index = 0
         while text:
             a, sep1, text = text.partition(self.left)
             if a:
-                for i, t, v in self.lang.get_tokens_unprocessed(a):
```
Comment (eric-wieser) [inline pygments/formatters/latex.py:470]: This wasn't safe - the nested tokenizer needs to see all the text at once, not just be served chunks of it at a time

**Review by Anteru — CHANGES_REQUESTED** [review]
Thanks! We have some merge conflicts due to the latest `yield from` improvements -- could you please take a look and update this as needed?

**Comment by eric-wieser** [comment]
Conflicts resolved, the changes in #1537 were irrelevant

**Comment by Anteru** [comment]
I did look at this and it all seems fine, but I don't quite understand the original code. Please bear with me -- I just need to convince myself it works (I'm fairly certain it does no harm :) ).

**Comment by eric-wieser** [comment]
I wouldn't spend too much time understanding the old code - it had both indentation mistakes and a bad design...

**Comment by Anteru** [comment]
Fair enough -- it does have tests now so I doubt it can be any worse than the original one; sorry for taking that long to merge.

**Pull request merged**